acme/directory.go

```diff
@@ -119,12 +125,26 @@ func (d *directory) getLinkFromBaseURL(typ Link, provisionerName string, abs boo
 	case FinalizeLink:
 		link = fmt.Sprintf("/%s/%s/%s/finalize", provisionerName, OrderLink.String(), inputs[0])
 	}
+
 	if abs {
-		baseURL := baseURLFromRequest
-		if baseURL == "" {
-			baseURL = "https://" + d.dns
+		// Copy the baseURL value from the pointer. https://github.com/golang/go/issues/38351
+		u := url.URL{}
+		if baseURL != nil {
+			u = *baseURL
+		}
```

Comment:
`url.URL` is usually used as a pointer, so it might be more comfortable to do `u := new(url.URL)`

Reply:
Doing this so as not to change the value inside baseURL.